src/core/eval.ml

```diff
@@ -179,8 +179,9 @@ and eq_modulo : ctxt -> term -> term -> bool = fun ctx a b ->
     fails,  the stack [stk] may be imperatively updated since a reduction step
     taken in elements of the stack is preserved (this is done using
     {!constructor:Terms.term.TRef}). *)
-and tree_walk : dtree -> ctxt -> stack -> (term * stack) option =
-  fun tree ctx stk ->
+and tree_walk :
+  dtree -> ctxt -> stack -> term option -> (term * stack) option =
```

Comment:
Thanks to update the comment  of tree_walk

Reply:
Actually, I don't understand what is this metatype argument. Could you please explain?

Reply:
this part has been removed 

the rationale was to fill an array in tree_walk with a precomputed type instead of a meta of type Kind (cf also comment below)